static/sharing.js

```diff
@@ -100,6 +100,11 @@ function initShareButton(getLink, layout) {
 
         getLinks(layout, function (theUrls) {
             urls = theUrls;
+            if (!urls[currentBind])
+                currentBind = 'Full';
+            root.find('.sources a').css("display", function () {
```

Comment:
Can you just use http://api.jquery.com/toggle/ here

Reply:
Can't see how I could use that? I don't want to do it based on previous state, I want to do it based on if there is data or not.

Reply:
`.toggle()` has an overload which takes a boolean, which decides if it will be displayed or not, regardless of the current state

Reply:
So you would prefer something like this?
```js
            root.find('.sources a').each(function () {
                $(this).toggle(!!urls[$(this).data().bind]);
            });
```

Reply:
I would prefer this method, as it leaves the css correct if the display is something other than block. (inline, inline-block) 

Although if we are just wanting to hide things that don't have a data-bind attribute, any reason it has to be done in javascript instead of just using a not selector in css?

Reply:
Now css is far from my area of expertise, but I can't see how a pure css solution would work. The data-bind attributes in the DOM just contains the _name_ of the sharing entry. And if that entry isn't there in the javascript object that contains the name -> value mapping for sharing entries (or rather if it has a false-like value) it should be hidden.

See how the code is
`$(this).toggle(!!urls[$(this).data().bind]);`
and not:
`$(this).toggle(!!$(this).data().bind);`


Maybe one could rework it so that it also has the value in data attributes in the DOM and get rid of the urls object in javascript side.  But that feels out of scope for this PR.


I have updated my branch to use foreach+toggle now.